/*
 * Copyright 2024 Sowers, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

option go_package = "bosca.io/api/protobuf/search";

import "google/api/annotations.proto";
import "content/metadata.proto";

package bosca.search;

service SearchService {
  rpc Search(SearchRequest) returns (SearchResponse) {
    option (google.api.http) = {
      post: "/v1/search"
      body: "*"
    };
  }
}

message SearchRequest {
  string query = 1;

  uint64 offset = 2;
  uint64 limit = 3;
}

message SearchResponse {
  repeated bosca.content.Metadata metadata = 1;
}